src/shogun/machine/EnsembleMachine.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+#ifndef __ENSEMBLEMACHINE_H
+#define __ENSEMBLEMACHINE_H
+
+#include <algorithm>
+#include <memory>
+#include <shogun/ensemble/CombinationRule.h>
+#include <shogun/features/Features.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/util/traits.h>
+#include <shogun/util/zip_iterator.h>
+#include <vector>
+namespace shogun
+{
+
+	class EnsembleMachine : public SGObject
+	{
+	public:
+		EnsembleMachine() = default;
+        
+		EnsembleMachine(const EnsembleMachine& other)
+		    : m_machines(other.m_machines)
+		{
+		}
```

Comment:
I would delete the copy constructor if possible, because you are doing shallow copies of the `Machine`s.

Reply:
Actually not sure... @vigsterkr should we enable copy constructors in `Machine` algorithms?